services/src/util/middleware.rs

```diff
@@ -0,0 +1,121 @@
+use std::future::{ready, Ready};
+
+use actix_web::{
+    dev::{forward_ready, Service, ServiceRequest, ServiceResponse, Transform},
+    http::header::{HeaderName, HeaderValue},
+    Error, HttpMessage,
+};
+use futures_util::future::LocalBoxFuture;
+use tracing::{event_enabled, Level};
+use tracing_actix_web::RequestId;
+
+const REQUEST_ID_HEADER: &str = "x-request-id";
+
+// There are two steps in middleware processing.
+// 1. Middleware initialization, middleware factory gets called with
+//    next service in chain as parameter.
+// 2. Middleware's call method gets called with normal request.
+pub struct OutputRequestId;
+
+// Middleware factory is `Transform` trait
+// `S` - type of the next service
+// `B` - type of response's body
+impl<S, B> Transform<S, ServiceRequest> for OutputRequestId
+where
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
+    S::Future: 'static,
+    B: 'static,
+{
+    type Response = ServiceResponse<B>;
+    type Error = Error;
+    type InitError = ();
+    type Transform = OutputRequestIdMiddleware<S>;
+    type Future = Ready<Result<Self::Transform, Self::InitError>>;
+
+    fn new_transform(&self, service: S) -> Self::Future {
+        ready(Ok(OutputRequestIdMiddleware {
+            service,
+            // Only send header if request_id gets logged, because
+            // otherwise correlation is impossible anyway.
+            enabled: event_enabled!(target: "tracing_actix_web::middleware", Level::WARN, request_id)
+                || cfg!(test),
+        }))
+    }
+}
+
+pub struct OutputRequestIdMiddleware<S> {
+    service: S,
+    enabled: bool,
+}
+
+impl<S, B> Service<ServiceRequest> for OutputRequestIdMiddleware<S>
+where
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
+    S::Future: 'static,
+    B: 'static,
+{
+    type Response = ServiceResponse<B>;
+    type Error = Error;
+    type Future = LocalBoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    forward_ready!(service);
+
+    fn call(&self, req: ServiceRequest) -> Self::Future {
+        // TODO: rethink this error handling
```

Comment:
remove todo and edit the expect-message why it should always work. something like: RequestId should be in extensions, because it was attached in xyz.

Reply:
[cc147b9](https://github.com/geo-engine/geoengine/pull/955/commits/cc147b9b31152d18b5705bf715d96e290064cc81)